src/FsAutoComplete.Core/SignatureFormatter.fs

```diff
@@ -42,7 +42,12 @@ module SignatureFormatter =
             if typ.IsTupleType || typ.IsStructTupleType then
                 let refTupleStr =
                     typ.GenericArguments
-                    |> Seq.map (formatFSharpType context)
+                    |> Seq.map
+                        (fun arg ->
+                            if arg.IsTupleType || arg.IsStructTupleType then
```

Comment:
Should this only happen on non-struct tuples? I guess that would be an unnecessary set of parens.

Reply:
struct tuple should get the text `struct(<inner tuple signature>)` if memory serves, to parallel the way they are created (ie `struct(1, 2)` would be `struct(int * int)`.

Reply:
Nice, thank you. Updated in ed15e9c.